DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIDatePickerRecorder.swift

```diff
@@ -0,0 +1,163 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+
+internal struct UIDatePickerRecorder: NodeRecorder {
+    private let wheelsStyleRecorder = WheelsStyleDatePickerRecorder()
+    private let compactStyleRecorder = CompactStyleDatePickerRecorder()
+    private let inlineStyleRecorder = InlineStyleDatePickerRecorder()
+
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
+        guard let datePicker = view as? UIDatePicker else {
+            return nil
+        }
+
+        guard attributes.isVisible else {
+            return InvisibleElement.constant
+        }
+
+        var nodes: [Node] = []
+
+        if #available(iOS 13.4, *) {
+            switch datePicker.datePickerStyle {
+            case .wheels:
+                nodes = wheelsStyleRecorder.recordNodes(of: datePicker, with: attributes, in: context)
+            case .compact:
+                nodes = compactStyleRecorder.recordNodes(of: datePicker, with: attributes, in: context)
+            case .inline:
+                nodes = inlineStyleRecorder.recordNodes(of: datePicker, with: attributes, in: context)
+            case .automatic:
+                // According to `datePicker.datePickerStyle` documentation:
+                // > "This property always returns a concrete style, never `UIDatePickerStyle.automatic`."
+                break
+            @unknown default:
+                nodes = wheelsStyleRecorder.recordNodes(of: datePicker, with: attributes, in: context)
+            }
+        } else {
+            // Observation: older OS versions use the "wheels" style
+            nodes = wheelsStyleRecorder.recordNodes(of: datePicker, with: attributes, in: context)
+        }
+
+        let isDisplayedInPopover: Bool = {
```

Comment:
Probably a style preference on my side, but it's a perfect candidate for nested function. IMO slightly more swiftly. Obviously not a blocker :)